aws/hll/aws-s3-transfer-manager/src/io/stream.rs

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::default::Default;
+use std::path::Path;
+
+use bytes::{Buf, Bytes};
+
+use crate::io::error::Error;
+use crate::io::path_body::PathBody;
+use crate::io::path_body::PathBodyBuilder;
+use crate::io::size_hint::SizeHint;
+
+/// Source of binary data.
+///
+/// `InputStream` wraps a stream of data for ease of use.
+#[derive(Debug)]
+pub struct InputStream {
+    pub(super) inner: RawInputStream,
+}
```

Comment:
What does `Input` mean here? This implies the existence of an `OutputStream` too, right?

Reply:
yeah it implies it but I don't think we'll implement it. Open to naming but essentially we don't want to use `ByteStream` as the input or output type for upload or download respectively (for different reasons). I'm not sure what we want to call them so open to bikeshedding.

`InputStream` is a common name for something you can read from which is what this is but I don't expect it to be re-used for both inputs and outputs the way `ByteStream` is.